apps/distgo/cmd/dist/rpmdist.go

```diff
@@ -45,6 +46,14 @@ func checkRPMDependencies() error {
 
 func rpmDist(buildSpecWithDeps params.ProductBuildSpecWithDeps, distCfg params.Dist, outputProductDir string, stdout io.Writer) (p Packager, rErr error) {
 	buildSpec := buildSpecWithDeps.Spec
+	osArchs := buildSpec.Build.OSArchs
+	expected := osarch.OSArch{OS: "linux", Arch: "amd64"}
+	if len(osArchs) != 1 || osArchs[0] != expected {
+		return nil, fmt.Errorf("RPM is only supported for %v", expected)
+	}
+	if err := checkRPMDependencies(); err != nil {
```

Comment:
I moved this from dist.go to rpmdist.go. This check should be performed by rpmdist since its specific to it.